Source/WebCore/Modules/highlight/Highlight.h

```diff
@@ -32,42 +32,60 @@
 
 namespace WebCore {
 
+class AbstractRange;
 class CSSStyleDeclaration;
 class DOMSetAdapter;
 class PropertySetCSSStyleDeclaration;
-class StaticRange;
 
-
-struct HighlightRangeData : RefCounted<HighlightRangeData>, public CanMakeWeakPtr<HighlightRangeData> {
-    HighlightRangeData(Ref<StaticRange>&& range)
-        : range(WTFMove(range))
+class HighlightRangeData : public RefCounted<HighlightRangeData>, public CanMakeWeakPtr<HighlightRangeData> {
+public:
+    static Ref<HighlightRangeData> create(Ref<AbstractRange>&& range)
     {
+        return adoptRef(*new HighlightRangeData(WTFMove(range)));
     }
-    static Ref<HighlightRangeData> create(Ref<StaticRange>&& range)
+
+    AbstractRange& range() const { return m_range.get(); }
+    const std::optional<Position>& startPosition() const { return m_startPosition; }
+    void setStartPosition(Position&& startPosition) { m_startPosition = WTFMove(startPosition); }
+    const std::optional<Position>& endPosition() const { return m_endPosition; }
+    void setEndPosition(Position&& endPosition) { m_startPosition = WTFMove(endPosition); }
+
+private:
+    explicit HighlightRangeData(Ref<AbstractRange>&& range)
+        : m_range(WTFMove(range))
     {
-        return adoptRef(*new HighlightRangeData(WTFMove(range)));
     }
-    Ref<StaticRange> range;
-    std::optional<Position> startPosition;
-    std::optional<Position> endPosition;
+
+    Ref<AbstractRange> m_range;
+    std::optional<Position> m_startPosition;
+    std::optional<Position> m_endPosition;
 };
 
 class Highlight : public RefCounted<Highlight> {
 public:
-    WEBCORE_EXPORT static Ref<Highlight> create(StaticRange&);
+    WEBCORE_EXPORT static Ref<Highlight> create(FixedVector<std::reference_wrapper<WebCore::AbstractRange>>&&);
```

Comment:
what's the difference between a FixedVector and an std::array?

Reply:
This is the type the bindings are passing us here so I cannot use a std::array. Unlike std::array, the length of a FixedVector is set at runtime, not compile time. std::array actually has the length as part of its type. FixedVector doesn't but the size doesn't change once constructed.

Reply:
ah, I see! thanks for the explanation!

Reply:
In other words, we wouldn't use a std::array here since we don't know at compile time how many ranges the JS will pass to the constructor.